Telegram/SourceFiles/base/assertion.h

```diff
@@ -43,10 +43,15 @@ inline constexpr void noop() {
 	std::abort();
 }
 
-inline constexpr void validate(bool condition, const char *message, const char *file, int line) {
+inline
+#if !defined(_MSC_VER) || (_MSC_VER > 1900)
+constexpr
+#endif
+  void validate(bool condition, const char *message, const char *file, int line) {
```

Comment:
I'm not getting what problem we're trying to solve with this change. Some version of MSVC doesn't support `constexpr void`? I suggest to drop support for that compiler; we do require a decent support of C++17.

Reply:
It's only one peace of code with MSVC 2015 Update 3:
```
inline constexpr void validate(bool condition, const char *message, const char *file, int line) {
(47)  (GSL_UNLIKELY(!(condition))) ? fail(message, file, line) : noop();
}
```
3>h:\tmp\kepka\telegram\sourcefiles\base/assertion.h(47): error C3249: illegal statement or sub-expression for 'constexpr' function

So we have decided to  use conditional compilation for only this function.